spec/unit/form_builder_spec.rb

```diff
@@ -318,14 +324,30 @@ def build_form(options = {}, form_object = Post.new, &block)
         end
       end
     end
-    it "should generate a nested text input once" do
-      expect(body).to have_selector("[id=post_author_attributes_first_name_input]", count: 1)
-      expect(body).to have_selector("[id=post_author_attributes_last_name_input]", count: 1)
-    end
-    it "should add author first and last name fields" do
-      expect(body).to have_selector("input[name='post[author_attributes][first_name]']")
-      expect(body).to have_selector("input[name='post[author_attributes][last_name]']")
+
+    include_examples :inputs_with_for_expectation
+
+  end
+
+  context "with two input fields 'for' at the beginning of block" do
+    let :body do
+      build_form do |f|
+        f.form_builder.instance_eval do
+          @object.author = User.new
+        end
+        f.inputs name: 'Author', for: :author do |author|
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.